presto-hive/src/main/java/com/facebook/presto/hive/ParquetRecordWriterUtil.java

```diff
@@ -85,7 +94,7 @@ public void write(Writable value)
                     throws IOException
             {
                 recordWriter.write(value);
-                length = fileWriter.getPos();
+                length = fileWriter.getPos() + columnWriteStore.getBufferedSize();
```

Comment:
I think it's better not to change the meaning of length, instead, the addition should be done in getWrittenBytes()